Source/MQTTnet/Packets/MqttConnectPacket.cs

```diff
@@ -20,10 +21,10 @@ public sealed class MqttConnectPacket : MqttPacket
 
         public string ClientId { get; set; }
 
-        public byte[] WillCorrelationData { get; set; }
+        public ReadOnlyMemory<byte> WillCorrelationData { get; set; }
 
         public ushort KeepAlivePeriod { get; set; }
-        
+
         public uint MaximumPacketSize { get; set; }
 
         public byte[] Password { get; set; }
```

Comment:
Isn't this also a candidate for ReadOnlyMemory replacement?

Reply:
Currently, it cannot be replaced with ReadOnlyMemory because when password is null, it means there is no password and Empty means there is a value. Its logic is consistent with string UserName.
Changing to ReadOnlyMemory affects the related Encode and Decode as well as unit tests. It is recommended to keep it as it is for now because it only affects the connect packet. This change can be completed through a new PR in the future.